mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -446,6 +459,11 @@ private[ml] object DefaultParamsReader {
     val cls = Utils.classForName(metadata.className)
     cls.getMethod("read").invoke(null).asInstanceOf[MLReader[T]].load(path)
   }
+
+  def loadInitialModel[M <: Model[M]](path: String, sc: SparkContext): Try[M] = {
+    val initialModelPath = new Path(path, "initialModel").toString
+    Try(loadParamsInstance[M](initialModelPath, sc))
```

Comment:
Note, now we always try to load the initial model, which I would guess adds some overhead in the cases when there is none, but we have to fail and catch the exception. Let's leave this for now since it's cleaner, and only change it if it becomes a problem.